src/PrestaShopBundle/Resources/config/services/core/form/form_handler.yml

```
@@ -124,6 +124,12 @@ services:
     arguments:
       - '@prestashop.core.form.identifiable_object.data_handler.order_message_form_data_handler'
 
+  prestashop.core.form.identifiable_object.handler.order_return_form_handler:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Handler\FormHandler'
+    factory: 'prestashop.core.form.identifiable_object.handler.form_handler_factory:create'
```

```suggestion
    factory: [ '@prestashop.core.form.identifiable_object.handler.form_handler_factory, 'create']
```

Since we are now using Symfony 4.4, you should use the new syntax for factories https://symfony.com/doc/4.4/service_container/factories.html#non-static-factories

Since this PR has been merged https://github.com/PrestaShop/PrestaShop/pull/25052 I believe this is the right way now, right @atomiix ?